metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/RestFunctions.java

```diff
@@ -0,0 +1,351 @@
+      Map<String, Object> globalRestConfig = (Map<String, Object>) globalConfig.get(STELLAR_REST_SETTINGS);
+      Map<String, Object> functionRestConfig = null;
+      if (args.size() > 1) {
+        functionRestConfig = getArg(1, Map.class, args);
+      }
+
+      // Add settings in order of precedence
+      RestConfig restConfig = new RestConfig();
+      if (globalRestConfig != null) {
+        restConfig.putAll(globalRestConfig);
+      }
+      if (functionRestConfig != null) {
+        restConfig.putAll(functionRestConfig);
+      }
+      return restConfig;
+    }
+
+    /**
+     * Returns the proxy HttpHost object if the proxy rest config settings are set.
+     * @param restConfig
+     * @return
+     */
+    protected Optional<HttpHost> getProxy(RestConfig restConfig) {
+      Optional<HttpHost> proxy = Optional.empty();
+      if (restConfig.getProxyHost() != null && restConfig.getProxyPort() != null) {
+        proxy = Optional.of(new HttpHost(restConfig.getProxyHost(), restConfig.getProxyPort(), "http"));
+      }
+      return proxy;
+    }
+
+    /**
+     * Builds the RequestConfig object by setting HttpClient settings defined in the rest config.
+     * @param restConfig
+     * @param proxy
+     * @return
+     */
+    protected RequestConfig getRequestConfig(RestConfig restConfig, Optional<HttpHost> proxy) {
+      RequestConfig.Builder requestConfigBuilder = RequestConfig.custom();
+      if (restConfig.getConnectTimeout() != null) {
+        requestConfigBuilder.setConnectTimeout(restConfig.getConnectTimeout());
+      }
+      if (restConfig.getConnectionRequestTimeout() != null) {
+        requestConfigBuilder.setConnectionRequestTimeout(restConfig.getConnectionRequestTimeout());
+      }
+      if (restConfig.getSocketTimeout() != null) {
+        requestConfigBuilder.setSocketTimeout(restConfig.getSocketTimeout());
+      }
+
+      proxy.ifPresent(requestConfigBuilder::setProxy);
+      return requestConfigBuilder.build();
+    }
+
+    /**
+     * Builds the HttpClientContext object by setting the basic auth and/or proxy basic auth credentials when the
+     * necessary rest config settings are configured.  Passwords are stored in HDFS.
+     * @param restConfig
+     * @param target
+     * @param proxy
+     * @return
+     * @throws IOException
+     */
+    protected HttpClientContext getHttpClientContext(RestConfig restConfig, HttpHost target, Optional<HttpHost> proxy) throws IOException {
+      HttpClientContext httpClientContext = HttpClientContext.create();
+      boolean credentialsAdded = false;
+      CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+      // Add the basic auth credentials if the rest config settings are present
+      if (restConfig.getBasicAuthUser() != null && restConfig.getBasicAuthPasswordPath() != null) {
+        String password = new String(readBytes(new Path(restConfig.getBasicAuthPasswordPath())), StandardCharsets.UTF_8).trim();
```

Comment:
Why are we calling `trim()`? Is whitespace not valid at the front/back of passwords?

Reply:
Just being defensive.  I found I had to do this when reading the contents from HDFS.  I can look into it again and try to track down exactly what the problem was.

Reply:
The issue was with my testing instructions.  I was unintentionally adding an extra newline at the end of the password file.  I removed trim() and added a warning to the README.  Will also update the test instructions.